loader_hub/opendal/base.py

```diff
@@ -0,0 +1,79 @@
+"""Opendal file and directory reader.
+
+A loader that fetches a file or iterates through a directory on AWS S3 or other compatible service.
+
+"""
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union
+import asyncio
+
+from llama_index import download_loader
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+import opendal
```

Comment:
actually these imports should probably be lazily loaded so that it doesn't break loading for other modules (otherwise unit tests also currently fail)